component/loki/source/kubernetes/kubernetes.go

```diff
@@ -168,28 +186,43 @@ func (c *Component) Update(args component.Arguments) error {
 		// No-op: manager already exists and options didn't change.
 	}
 
-	// Convert input targets into targets to give to tailer.
-	targets := make([]*kubetail.Target, 0, len(newArgs.Targets))
+	c.resyncTargets(newArgs.Targets)
+	c.args = newArgs
+	return nil
+}
+
+func (c *Component) resyncTargets(targets []discovery.Target) {
+	distTargets := discovery.NewDistributedTargets(c.args.Clustering.Enabled, c.cluster, targets)
+	targets = distTargets.Get()
 
-	for _, inTarget := range newArgs.Targets {
-		lset := inTarget.Labels()
+	tailTargets := make([]*kubetail.Target, 0, len(targets))
+	for _, target := range targets {
+		lset := target.Labels()
 		processed, err := kubetail.PrepareLabels(lset, c.opts.ID)
 		if err != nil {
 			// TODO(rfratto): should this set the health of the component?
```

Comment:
In what situations will PrepareLabels error? Is there a case where we discard a target and increment a counter or something?

Reply:
Looks like it fails based on validations:

> // Validation of lset fails if there is no label indicating the pod namespace,
> // pod name, or container name.

In this particular case, the component would've been given a bad target.